python/lsst/ts/ess/labjack/labjack_accelerometer_data_client.py

```diff
@@ -422,17 +486,54 @@ async def callback(
 
         try:
             psd = np.abs(np.fft.rfft(np.array(scaled_data))) ** 2
-            psd_kwargs = {
-                f"accelerationPSD{axis}": psd[i, self.psd_start_index :]
-                for i, axis in enumerate(("X", "Y", "Z"))
-            }
-            await self.topic.set_write(
-                interval=self.sampling_interval,
-                minPSDFrequency=self.psd_frequencies[self.psd_start_index],
-                maxPSDFrequency=self.psd_frequencies[-1],
-                numDataPoints=self.config.num_frequencies,
-                **psd_kwargs,
-            )
+            for i, accelerometer in enumerate(self.accelerometers):
+                channel_start_index = i * NUM_CHANNELS_PER_ACCELEROMETER
+
+                # Write scaled acceleration data. This usually requires
+                # multiple messages per accelerometer, because the arrays
+                # are rather.
+                accel_data_start_index = 0
+                done_writing_accel = False
+                while not done_writing_accel:
+                    num_accel_to_write = self.num_samples - accel_data_start_index
+                    if num_accel_to_write > self.accel_array_len:
+                        num_accel_to_write = self.accel_array_len
+                    else:
+                        done_writing_accel = True
+
+                    accel_kwargs = {
+                        f"acceleration{axis}": scaled_data[
+                            channel_start_index + channel_offset,
+                            accel_data_start_index : accel_data_start_index
+                            + num_accel_to_write,
+                        ]
+                        for channel_offset, axis in enumerate(("X", "Y", "Z"))
+                    }
+                    accel_data_start_index += num_accel_to_write
+                    await self.accel_topic.set_write(
```

Comment:
Writing raw data should be optional. Please can you make that configurable? PSD data shall be produced all the time, raw on request when configured.

Reply:
I am not sure @patrickingraham considers this optional, but I will add the parameter and default it to true.

Reply:
Done (after checking with Patrick; I'm very glad you both agreed that this is wanted). I made it a required parameter because I did not feel that a default made much sense.